### DISCUSSION TITLE
How to hide MKDocs TOC for just one page

### DISCUSSION THREAD

**Discussion by ghost** [mkdocs/mkdocs#3814] [Q&A] [answered]
I have this really styled front page, but theres the TOC render which is an issue cus it looks really ugly. Is there any way to get rid of the TOC for just that one page?

**Comment by pawamoy** [comment]
Given you've asked about landing pages on Material for Mkdocs (https://github.com/squidfunk/mkdocs-material/discussions/7390), I suppose you're using the `mkdocs-material` theme, and therefore that would be a question to ask on their tracker. But what you want to do is actually documented on their site: https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/?h=hide#hiding-the-sidebars. Closing!

**Comment by ghost** [comment]
No I'm asking for the MKDocs theme, my documentation does not use Material for MKDocs(the one that I am working on)

**Comment by pawamoy** [comment, marked as answer]
In that case, you might try to hide with CSS! It looks like MkDocs itself is doing just that:

```css
body.homepage>div.container>div.row>div.col-md-3 {
    display: none;
}
```

:slightly_smiling_face:

**Comment by ghost** [comment]
alright thanks. Ill try this later given i dont have time right now

**Comment by pawamoy** [comment]
I'll move this to Q/A, and if CSS doesn't work for you, feel free to open a new feature request!

**Comment by kwameopareasiedu** [comment]
Here's how I implemented it in a reusable way. Let's say I wanted to hide TOC in my `index.md`.

1. Add some non-standard HTML tags to the file you'd like to hide markdown from. Since these tags will not be rendered, I use them as markers. In this example, I used `<hide_toc>`:

    `index.md`
    ```markdown
    # Hello World

    Here's my awesome site
    <hide_toc>
    ```

2. Create a custom javascript file with the following code:
    `js/hide_toc.js`
    ```javascript
    const hasHideTocTag = !!document.querySelector("hide_toc");

    if (hasHideTocTag) {
      const containerRow = document.querySelector("body > .container > .row");
      const tocCol = containerRow.querySelector("& > div:first-child");
      tocCol.style.visibility = "collapse";
    }
    ```

3. Register the custom javascript file in your `mkdocs.yml`
    `mkdocs.yml`
    ```yaml
    site_name: Site
    extra_javascript:
        - js/hide_toc.js
    ```

Now for every other page you want to hide TOC, just add `<hide_toc>` anywhere in the file.
